timemachine/ff/handlers/bonded.py

```diff
@@ -73,9 +73,20 @@ class HarmonicBondHandler(ReversibleBondHandler):
     @staticmethod
     def static_parameterize(params, smirks, mol):
         mol_params, bond_idxs = super(HarmonicBondHandler, HarmonicBondHandler).static_parameterize(params, smirks, mol)
+
+        # validate expected number of bonds
+        expected_num_bonds = mol.GetNumBonds()
+        assigned_num_bonds = len(bond_idxs)
+        if assigned_num_bonds != expected_num_bonds:
```

Comment:
strictly speaking, we probably don't want to just assert that the # of bonds are the same, but that their connectivities are the same

```
rd_bonds = set()
for b in mol.GetBonds():
    rd_bonds.add(tuple(sorted([b.GetBeginAtomIdx(), b.GetEndAtomIdx()])))

ff_bonds = set()
for i,j in bond_idxs:
        rd_bonds.add(tuple(sorted([i,j])))

assert rd_bonds == ff_bonds
```

Reply:
Good call! Applied in https://github.com/proteneer/timemachine/pull/742/commits/10153916345f8342e202f9bc2ad16a8e2b87df86